dae/dae/pedigrees/family.py

```diff
@@ -330,6 +330,39 @@ def to_json(self):
             "tags": self.tags
         }
 
+    def get_columns(self):
+        column_names = set(self.members_in_order[0]._attributes.keys())
+        columns = [
+            col
+            for col in PEDIGREE_COLUMN_NAMES.values()
+            if col in column_names
+            or col in ["generated", "not_sequenced"]
+        ]
+        extension_columns = column_names.difference(set(columns))
+        extension_columns = extension_columns.difference(
+            set(["sample_index"])
+        )
+        columns.extend(sorted(extension_columns))
+        return columns
+
+    def to_rows(self, sep="\t", columns=None):
```

Comment:
I really hate the name `to_rows` - it is so general and vague. This method does something very, very specific - it serializes the object to a list of string, each string having the attributes of a member tab separated.

Personally I think you should move this method (and the to_columns method) to a seprate class, let's say `FamilyStreamTsvSerializer` that has a method serialize that returns what to_rows returns now.

Reply:
fixed

Reply:
fixed